src/fontra/views/fontinfo/panel-sources.js

```diff
@@ -1,14 +1,604 @@
+    return {
+      general: {
+        name: source.name,
+        italicAngle: source.italicAngle ? source.italicAngle : 0,
+        //isSparce: source.isSparce ? source.isSparce : false,
+      },
+      location: { ...source.location },
+      verticalMetrics: this.prepareVerticalMetricsForController(source.verticalMetrics),
+      // TODO: hhea, OS/2 verticalMetrics, etc
+      // customData: { ...source.customData },
+    };
+    // NOTE: Font guidlines could be read/write here,
+    // but makes more sense directly in the glyph editing window.
+  }
+
+  prepareVerticalMetricsForController(verticalMetrics) {
+    let newVerticalMetrics = {};
+    for (const key in verticalMetrics) {
+      newVerticalMetrics[`value-${key}`] = verticalMetrics[key].value;
+      newVerticalMetrics[`zone-${key}`] = verticalMetrics[key].zone;
+    }
+    return newVerticalMetrics;
+  }
+
+  checkSourceLocation(axisName, value) {
+    const newLocation = { ...this.source.location, [axisName]: value };
+    return this.checkSourceEntry("location", undefined, newLocation);
+  }
+
+  checkSourceEntry(key, valueKey = undefined, value) {
+    let errorMessage = "";
+    for (const sourceIdentifier in this.sources) {
+      if (sourceIdentifier == this.sourceIdentifier) {
+        // skip the current source
+        continue;
+      }
+      const source = this.sources[sourceIdentifier];
+
+      let existsAlready = false;
+      let sourceValue;
+
+      if (valueKey == undefined) {
+        if (key == "location") {
+          sourceValue = locationToString(source[key]);
+          value = locationToString(value);
+        } else {
+          sourceValue = source[key];
+        }
+      } else {
+        sourceValue = source[key][valueKey];
+      }
+
+      if (sourceValue == value) {
+        existsAlready = true;
+      }
+
+      if (existsAlready) {
+        errorMessage = `${key}${valueKey ? " " + valueKey : ""}: “${value}”
+          exists already, please use a different value.`;
+        break;
+      }
+    }
+
+    if (errorMessage) {
+      message(`Can’t edit font source`, errorMessage);
+      this.setupUI();
+      return false;
+    }
+    return true;
+  }
+
+  editSource(editFunc, undoLabel) {
+    const root = { sources: this.sources };
+    const changes = recordChanges(root, (root) => {
+      editFunc(root.sources[this.sourceIdentifier]);
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+    }
+  }
+
+  deleteSource() {
+    const undoLabel = `delete source '${this.source.name}'`;
+    const root = { sources: this.sources };
+    const changes = recordChanges(root, (root) => {
+      delete root.sources[this.sourceIdentifier];
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+
+  toggleShowHide() {
+    const el = this.querySelector("#open-close-icon");
+    el.style.transform =
```

Comment:
Please use the following tactic (used in the accordion view, and elsewhere):
- instead of changing the style via code, you add or remove a (well named) class
- then, in the CSS you can select by that class, and apply the appropriate changes

This is the better and cleaner way, as it keeps the actual styling near the other styles instead of embedded in the code.

Reply:
I reworked the toggleShowHide and all css dependencies. I make this as done.